vector2/vector2.go

```diff
@@ -241,10 +286,16 @@ func (v Vector[T]) X() T {
 	return v.x
 }
 
-// SetX changes the x component of the vector
-func (v Vector[T]) SetX(newX T) Vector[T] {
+func (v Vector[T]) SetX(X T) Vector[T] {
+	return Vector[T]{
+		x: X,
+		y: v.y,
+	}
+}
+
+func (v Vector[T]) Dx(dX T) Vector[T] {
```

Comment:
Let's keep using `Add` as it's already used by this lib.

This goes for all other uses of D or Delta

```suggestion
func (v Vector[T]) AddX(dX T) Vector[T] {
```